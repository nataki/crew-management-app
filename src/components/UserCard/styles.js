export default {
    user_card: {
        border: '1px solid grey',
        boxShadow: '0 4px 8px 0 rgba(0,0,0,0.2)',
        borderRadius: '3px',
        marginBottom: '10px',
    },
    user_info: {
        display: 'flex',
    },
    user_photo: {
        marginRight: '10px',
        padding: '5px'
    },
    user_name: {
        '& div': {
            fontSize: '0.8em',
            fontStyle: 'italic',
            marginTop: '10px'
        }
    },
    controls: {
        display: 'flex',
        justifyContent: 'space-between',
        padding: '5px'
    }
}