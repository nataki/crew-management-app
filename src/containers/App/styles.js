export default {
    app: {
    },
    header: {
        display: 'flex'
    },
    container: {
        display: 'flex'
    }
}