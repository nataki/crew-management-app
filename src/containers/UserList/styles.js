export default {
    column: {
        flex: '1',
        padding: '20px',
    },
    category_title: {
        fontWeight: 'bold',
        lineHeight: '40px',
        textAlign: 'center'
    }
}