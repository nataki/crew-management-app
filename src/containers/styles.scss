.app {
    .header{
        display: flex;
    }
    .container{
        display: flex;
        .column {
            flex: 1;
            padding: 20px;

            .category_title{
                text-align: center;
                font-weight: bold;
                line-height: 40px;
            }
        }
    }
}

.user_card {
    border: 1px solid grey;
    box-shadow: 0 4px 8px 0 rgba(0,0,0,0.2);
    border-radius: 3px;
    margin-bottom: 10px;
    .user_info {
        display: flex;
        .user_photo {
            margin-right: 10px;
            padding: 5px;
        }
        .user_name div {
            font-size: 0.8em;
            font-style: italic;
            margin-top:10px;
        }
        .controls {
            display: flex;
            justify-content: space-between;
            padding: 5px;
        }
    }
}