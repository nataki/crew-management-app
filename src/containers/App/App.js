/* @flow */

import React, {Component} from "react";
import {connect} from "react-redux";
import injectSheet from 'react-jss';
import {compose} from 'recompose';

import styles from "./styles";
import FilterForm from "../FilterForm";
import UserList from "../UserList";
import {fetchUsersAction} from "../../actions";
import {STATUS_LOADING} from "../../constants";

type Props = {
    classes: { app: string, header: string, container: string },
    status: string,
    fetchUsers: () => void,
};

export class App extends Component<Props> {

    static defaultProps = {
        classes: {},
        status: STATUS_LOADING,
        fetchUsers: () => {},
    };

    componentDidMount() {
        this.props.fetchUsers();
    }

    render() {
        const {
            classes,
            status,
        } = this.props;

        return (
            <div className={classes.app}>
                <div className={classes.header}>
                    <FilterForm />
                </div>
                <div className={classes.container}>
                    {status === STATUS_LOADING ? (
                        <h2>Loading...</h2>
                    ) : (
                        <UserList />
                    )}
                </div>
            </div>
        );
    }
}

const mapStateToProps = (state) => ({
    status: state.status,
});
const mapDispatchToProps = dispatch => ({
    fetchUsers: () => dispatch(fetchUsersAction()),
});

export default compose(
    injectSheet(styles),
    connect(mapStateToProps, mapDispatchToProps)
)(App);