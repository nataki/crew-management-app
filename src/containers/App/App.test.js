import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import {App} from './App';
import styles from "./styles";

//override generated classnames
const classes = Object.keys(styles)
    .reduce(
        (prev, curr) => ({
            ...prev,
            [curr]: curr
        }),
        {}
    );

describe('App - Snapshot',() => {
    it('capturing Snapshot of App', () => {
        const renderer = new ShallowRenderer();
        renderer.render(<App />);
        const renderedValue = renderer.getRenderOutput();
        expect(renderedValue).toMatchSnapshot();
    });
});