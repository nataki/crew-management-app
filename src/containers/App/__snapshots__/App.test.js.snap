// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App - Snapshot capturing Snapshot of App 1`] = `
<div>
  <div>
    <Connect(FilterForm) />
  </div>
  <div>
    <h2>
      Loading...
    </h2>
  </div>
</div>
`;
